### PR TITLE
NMS-12397: Refactoring of the Cassandra installation instructions

### PR DESCRIPTION
**Pull request by mfuhrmann**
### All Contributors

* [x] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [x] Have you made a comment in that issue which points back to this PR?
* [x] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [x] If this is a new or updated feature, is there documentation for the new behavior?
* [x] If this is new code, are there unit and/or integration tests?
* [x] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

### Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

Once the reviewer(s) accept the PR and the branch passes continuous integration in Bamboo, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-12397
* Bamboo (Continuous Integration): https://bamboo.opennms.org/

### CODE REVIEW

**Comment by mfuhrmann** [comment]
I removed our install steps since the apache.org guide already contains the complete procedure.
I only added the autostart and apt mark hold part as recommendation.

**Comment by indigo423** [comment]
IMHO, I would remove the Cassandra install guide for *all* distributions not just for Debian and add a generic reference with something like ... install Cassandra from the official installation  instructions.

**Comment by christianpape** [comment]
> IMHO, I would remove the Cassandra install guide for all distributions not just for Debian and add a generic reference with something like ... install Cassandra from the official installation instructions.

I agree.

**Comment by indigo423** [comment]
I've reworked the Newts setup guide and removed the Cassandra installation instructions with references to Cassandra so we don't have to maintain it. I'll rewrite the JIRA issue as well so it reflects the refactoring of this change better.

**Comment by indigo423** [comment]
@agalue I've worked in your comment and instead of using the newts init command, I've used your CQL statement to initialize the keyspace. I've shortened a bit the explanation in the first section why we recommend to use TWCS instead of STSC to avoid adding a wall of text :) What do you think?

Thanks a lot for adding this input here.

File: opennms-doc/guide-install/src/asciidoc/text/time-series-storage/newts.adoc [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+EOF
+----
+
+<1> Associate time series data by the foreign ID instead the database generated node ID
+<2> Set time series strategy to use `newts`
+<3> Host or IP addresses of the _Cassandra_ cluster nodes, can be a comma separated list
+<4> Name of the keyspace which is initialized and used
+<5> Port to connect to _Cassandra_
+
+===== Step 2: Initialize the _Newts_ schema in _Cassandra_
```
Comment (j-white) [inline opennms-doc/guide-install/src/asciidoc/text/time-series-storage/newts.adoc]: If this is what we're recommending to use for the default schema, should we not just update `newts init` to use this instead?

Comment (agalue) [reply opennms-doc/guide-install/src/asciidoc/text/time-series-storage/newts.adoc]: That's another valid approach. I envision that particular enhancement as follow:

Request not only the replication factor but also the desired retention (TTL), from there, calculate the window settings accordingly and generate the schema. Now, I think that would be in the Newts code base, right? But, without having those changes in place, the documentation should be something useful to have.

**Comment by agalue** [comment]
@indigo423 It is STCS (Size Tiered Compaction Strategy), not STSC ;)

File: opennms-doc/guide-install/src/asciidoc/text/time-series-storage/newts.adoc (hunk 1 above)
Comment (indigo423) [reply opennms-doc/guide-install/src/asciidoc/text/time-series-storage/newts.adoc]: > If this is what we're recommending to use for the default schema, should we not just update `newts init` to use this instead?

that would be my next issue and investigate how to change it there :)

**Comment by indigo423** [comment]
@agalue Fixed the TODO and @j-white I've created an enhancement issue with our findings in [NEWTS-109](https://issues.opennms.org/browse/NEWTS-109).

**Comment by indigo423** [comment]
I've dug a little bit deeper and I think we should solve the problem @agalue mentioned in NEWTS-109 instead of here. To address the STCS vs. TWCS issue, I've added a disclaimer to explain this is not a production-ready tuned setup and referenced to @agalue article which describes a lot of things in [greater detail](https://opennms.discourse.group/t/sizing-cassandra-for-newts/771).

Any comments?

**Comment by indigo423** [comment]
> LGTM!
> 
> Maybe it worth updating the discourse article to include how to set up the whole newts keyspace instead of focusing on how to tune just the samples table for TWCS, now that those details are not here anymore. What do you think?

I made the Discourse article a Wiki post so people can maintain it. I'll add the instructions in the discourse article.

**Pull request merged**